core/components/atoms/_chip/index.tsx

```diff
@@ -82,49 +100,67 @@ export const GenericChip = (props: GenericChipProps) => {
           <Text data-test="DesignSystem-GenericChip--Text" color={textColor} className="Chip-text">
             {label}
           </Text>
-        </>
+        </div>
       );
     }
     return label;
   };
 
+  const tooltipText = () => {
+    const labelText = typeof label === 'string' ? label : '';
+
+    if (labelPrefix) {
+      return `${labelPrefix} ${labelText}`;
+    }
+    return labelText;
+  };
+
   return (
-    // TODO(a11y)
-    // eslint-disable-next-line
-    <div
-      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-      tabIndex={disabled ? -1 : 0}
-      data-test="DesignSystem-GenericChip--GenericChipWrapper"
-      {...baseProps}
-      className={`Chip-wrapper ${className}`}
-      onClick={onClickHandler}
-    >
-      {icon && (
-        <Icon
-          data-test="DesignSystem-GenericChip--Icon"
-          name={icon}
-          type={iconType}
-          appearance={iconAppearance('left')}
-          className={iconClass('left')}
-        />
-      )}
-      {renderLabel()}
-      {clearButton && (
-        <div
-          role="button"
-          onClick={onCloseHandler}
-          tabIndex={disabled ? -1 : 0}
-          onKeyDown={onKeyDownHandler}
-          className={iconClass('right')}
-          data-test="DesignSystem-GenericChip--clearButton"
-        >
-          <Icon name="clear" appearance={iconAppearance('right')} className="p-2" />
-        </div>
-      )}
+    <div>
+      <Tooltip showTooltip={isTextTruncated} data-test="DesignSystem-GenericChip--Tooltip" tooltip={tooltipText()}>
+        <>
```

Comment:
Please fix this `DesignSystem-GenericChip--Tooltip` data-test as in this case tooltip does not act as a modifier

Reply:
@anuradha9712 can you please suggest me what should i use here as in this file i see we are using this pattern only to pass the value to the data-test i.e, `DesignSystem-ComponentName--ElementDescription`.